src/praisonai-train/praisonai_train/data/qc.py

```
@@ -0,0 +1,98 @@
+            g = ngrams(fields(r)[0])
+            if any(jaccard(g, g2) > near_j for g2 in sigs[-3000:]):
+                drops["near_dup"] += 1
+                continue
+            sigs.append(g)
+            deduped.append(r)
+        kept = deduped
+
+    lens = [len((fields(r)[2] or "").split()) for r in kept]
+    cv = st.pstdev(lens) / max(st.mean(lens), 1) if lens else 0.0
+    toks = [t for r in kept for t in norm(fields(r)[0]).split()]
+    bigrams = list(zip(toks, toks[1:]))
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)